lib/influxdb/client/http.rb

```diff
@@ -84,8 +85,18 @@ def resolve_error(response)
       raise InfluxDB::Error, response
     end
 
-    def handle_successful_response(response, options)
-      parsed_response = JSON.parse(response.body) if response.body
+    def handle_successful_response(response, options, json_streaming)
+      if json_streaming
+        if response.body
+          parsed_response = {}
+          response.body.each_line do |line|
+            parsed_response.merge!(JSON.parse(line)) { |_key, old, new| old + new }
+          end
+        end
+      elsif response.body
+        parsed_response = JSON.parse(response.body)
+      end
```

Comment:
Here, I'd change quite a bit.

1. instead of `foo={}; bar.each{ manipulate(foo) }; return foo`, follow a more functional approach
2. avoid naming variables "new"

How about this?

```
if options.fetch(:json_streaming, false)
  parsed_response = response.body.each_line.with_object({}) {|line, parsed| # (1)
    parsed.merge!(JSON.parse(line)) {|_key, oldval, newval|
      oldval + newval # (2)
    }
  }
end
```